src/datasets/packaged_modules/json/json.py

```diff
@@ -109,84 +110,79 @@ def _cast_table(self, pa_table: pa.Table) -> pa.Table:
             pa_table = table_cast(pa_table, self.config.features.arrow_schema)
         return pa_table
 
-    def _generate_tables(self, files):
-        for file_idx, file in enumerate(itertools.chain.from_iterable(files)):
-            # If the file is one json object and if we need to look at the items in one specific field
+    def _generate_tables(self, files: List[str]) -> Generator:
+        for file_idx, file in enumerate(files):
```

Comment:
why assuming that files is a list of strings ? it isn't

and overall this PR is pretty hard to review since you're doing a lot if small and unnecessary changes, I'd suggest opening a new PR and try to do minimal changes instead